apps/devhub/views.py

```diff
@@ -1735,3 +1729,26 @@ def docs(request, doc_name=None):
 def search(request):
     query = request.GET.get('q', '')
     return render(request, 'devhub/devhub_search.html', {'query': query})
+
+
+def api_key_agreement(request):
+    next_step = reverse('devhub.api_key')
+    return render_agreement(request, 'devhub/api/agreement.html', next_step)
+
+
+def render_agreement(request, template, next_step, step=None):
+    if request.method == 'POST':
+        request.user.update(read_dev_agreement=now())
+        return redirect(next_step)
+
+    if request.user.read_dev_agreement is None:
+        return render(request, template,
+                      {'step': step})
+    else:
+        response = redirect(next_step)
```

Comment:
Is the developer submission flow working as expected after this change? If it's not, we should fix it.